### DISCUSSION TITLE
Объёмы и формат практики

### DISCUSSION THREAD

**Discussion by pepelsbey** [doka-guide/content#1804] [Разное]
У нас есть блоки «В работе», которые выводятся после основого контента. Они оформляются так:

```html
<h2>В работе</h2>
<h3>имя_автора</h3>
```

И дальше идёт то, что решил посоветовать автор.

Так вышло, что некоторые советы имеют объёмы больше, чем статьи. И во многих есть заголовки, обычно второго уровня. Например:

```html
<h2>В работе</h2>
<h3>@KognitivnayaBuena</h3>
<h2>XML vs HTML vs JSON</h2>
<h3>XML</h3>
```

И это конечно же сводит с ума боковую навигацию:

<img width="234" alt="image" src="https://user-images.githubusercontent.com/105274/136620182-ac2e9822-2b2d-4e3e-b7d9-2778e9e87ee9.png">

На этой картинке всё, что идёт после `@KognitivnayaBuena` должно быть вложено туда.

Список советов с заголовками:

- https://doka.guide/tools/xml/#kognitivnayabuena
- https://doka.guide/tools/web-server/#igsekor
- https://doka.guide/tools/tcp-udp-protocols/#igsekor
- https://doka.guide/tools/nginx-web-server/#igsekor
- https://doka.guide/tools/network/#igsekor
- https://doka.guide/tools/file/#igsekor

**Comment by pepelsbey** [comment]
Предлагаю следующее:

1. Трансформировать существующие советы с заголовками (всего 6 из 312) в статьи: включить в существующие или завести новые
2. Написать правила создания советов, которые описывают их жанр, объём и регулируют, что там нет заголовков.

Это поможет в следующем:

1. Решит проблему с сайдбаром и вложенностью заголовков
2. Подскажет формат для будущих авторов
3. Сделает структуру статей более понятной: сейчас у нас после статьи начинаются другие статьи поменьше

**Comment by solarrust** [comment]
Я бы била только по именам авторов и игнорила вложенные заголовки

**Reply by pepelsbey** [reply]
А что делать с объёмными статьями после статей? Если из них просто выкинуть заголовки, они начинают слипаться в очень-очень длинный совет. Я добавил выше список с примерами.

**Reply by igsekor** [reply]
Может и бить на несколько практических примеров?

**Comment by igsekor** [comment]
Не хочется добавлять переменных, но что, если советов несколько у одного и того же автора. Если уменьшать объём, то подобные ситуации встречаться будут. Это можно контролировать, но обосновать не просто для авторов. Это может привести в лёгкому разочарованию, а это нам, кажется, не надо.

**Comment by igsekor** [comment]
> Список советов с заголовками:
> 
> * https://doka.guide/tools/xml/#kognitivnayabuena
> * https://doka.guide/tools/web-server/#igsekor
> * https://doka.guide/tools/tcp-udp-protocols/#igsekor
> * https://doka.guide/tools/nginx-web-server/#igsekor
> * https://doka.guide/tools/network/#igsekor
> * https://doka.guide/tools/file/#igsekor

5 статей моих) Так что порешаем быстро :)

**Comment by tachisis** [comment]
Я бы не писала правила, а решала в каждом индивидуальном случае — выделять ли это уже в отдельную статью или нет. Потому что если мы ограничим авторов советов на входе, они могут и не выдать много важного материала. Проще отщипнуть и пересадить потом.

**Comment by nlopin** [comment]
Кажется в большинстве случаев это что-то типа раздела «Практика». Я бы внёс их в текст основной статьи с соовтветсвующим заголовком.
